### PR TITLE
Cache file upload original names

### PR DESCRIPTION
**Pull request by MartinRiese**
## Product Description

When deleting a repeat group we need to rebuild all the following groups because the IDs are reassigned by formplayer. Unfortunately FP does not keep track of the original filename automatically and only sends the uuid based generated filename for the upload in the delete-repeat response. To be able to still display the original filename we store it in a map with the generated name as the key. That way we can figure out what it is when we rebuild that part of the UI.

## Technical Summary

https://dimagi.atlassian.net/browse/USH-5078

## Safety Assurance

tested locally and on staging

### Automated test coverage

corehq/apps/cloudcare/static/cloudcare/js/form_entry/spec/form_ui_spec.js

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by Robert-Costello — APPROVED** [review]
This looks good to me. Not blocking, but is it possible to add a test/assert that checks the filename is cached and/or reassigned after a repeat is deleted?

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js [hunk 1]
```diff
@@ -948,17 +949,26 @@ hqDefine("cloudcare/js/form_entry/entries", [
                 self.question.formplayerMediaRequest = $.Deferred();
                 self.cleared = false;
             }
-            var fixedNewValue = newValue.replace(constants.FILE_PREFIX, "");
+            const fixedNewValue = newValue.replace(constants.FILE_PREFIX, "");
             self.fileNameDisplay(fixedNewValue);
             self.answer(fixedNewValue);
+        } else if (cachedFilename) {
```
Comment (Jtang-1) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js:955]: Do you know what the code above is doing? What situation is `newValue` an empty string vs `constants.NO_ANSWER` (null)?

```
        if (newValue === "" && self.question.filename) {
            self.question.hasAnswered = true;
            self.fileNameDisplay(self.question.filename());
   ```
It looks like Minha added this previously [here](https://github.com/dimagi/commcare-hq/pull/35041/files) to solve a similar issue with the file question being rebuilt for question tiles. I'm not understanding why that doesn't solve the same issue here.

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js [hunk 2]
```diff
@@ -938,7 +938,8 @@ hqDefine("cloudcare/js/form_entry/entries", [
     FileEntry.prototype = Object.create(EntrySingleAnswer.prototype);
     FileEntry.prototype.constructor = EntrySingleAnswer;
     FileEntry.prototype.onPreProcess = function (newValue) {
-        var self = this;
+        const self = this;
+        const cachedFilename = self.question.form().fileNameCache[self.answer()];
```
Comment (Jtang-1) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js:942]: what's the distinction between `self.answer()` and `self.question.answer()`?

Comment (MartinRiese) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js:942]: There is none. self.answer just references to self. question.answer

**Review by Robert-Costello — APPROVED** [review]

**Review by minhaminha — APPROVED** [review]

**Pull request merged**